src/lib.rs

```diff
@@ -468,7 +473,43 @@ impl Inotify {
             }
         };
 
-        Ok(Events::new(Rc::downgrade(&self.fd), buffer, num_bytes))
+        Ok(Events::new(Arc::downgrade(&self.fd), buffer, num_bytes))
+    }
+
+    /// Create a stream which collects events
+    ///
+    /// Returns a `Stream` over all events that are available. This stream never is an
+    /// infinite source of events.
+    ///
+    /// An internal buffer which can hold the maximum possible size is used.
+    pub fn event_stream(mut self, handle: &Handle) -> io::Result<impl Stream<Item = EventOwned, Error = io::Error>> {
+        self.close_on_drop = false;
+
+        // Pass ownership of the file descriptor off.
+        let file = unsafe { File::from_raw_fd(*self.fd) };
+        let file = tokio_file_unix::File::new_nb(file)?
+            .into_reader(handle)?;
+
+        // Use this when 1.24 is available for use. We can use the hard-coded 16 due to Linux's ABI
+        // guarantees.
+        // const EVENT_MAX_SIZE: usize = mem::size_of::<ffi::inotify_event>() + (FILENAME_MAX as usize) + 1;
+        const EVENT_MAX_SIZE: usize = 16 + (FILENAME_MAX as usize) + 1;
+
+        Ok(length_delimited::Builder::new()
+            .little_endian()
+            // FIXME: See https://github.com/tokio-rs/tokio/pull/144
+            // .native_endian()
```

Comment:
This method doesn't exist yet. See tokio-rs/tokio#144.

Reply:
I don't want to merge this as-is. Do you think it's worth waiting for your PR to be merged? If not we can temporarily duplicate the functionality here.

Reply:
I didn't expect it to; it does have a `WIP` prefix on the commit message :) .

I'd like to know what the tokio developers think about it at least before reimplementing here.

Reply:
Sorry, I realize this is WIP and didn't mean to imply that you wanted to get it merged like this. I just wanted to make my opinion explicit, but should have worded that better.

Okay then, let's wait what happens with your tokio PR before proceeding here.